### ISSUE TITLE
Launchpad 2.0 not working (lumira designer 2.0)

### ISSUE THREAD

**Issue by dcm4you** [org-scn-design-studio-community/lumiradesignercommunityext#6]
Hello,

In DS < 2.0 we created a launchpad showcase (still with version 1.0). In Lumira 2.0 this isn't working any more. We tried launchpad 2.0 but the same issue occurs. There are no Tiles which are created. I also don't get any error messages. How to fix this?

Version: SAP BusinessObject Lumira Designer 20.2.0
 SCN Lumira Designer 2.0 SDK Components

With regards

Dennis

**Comment by guvenv** [comment]
i have the same problem!

**Comment by hakeemma** [comment]
Any update on this issue

**Comment by giannarelis** [comment]
Quite critical for our deployment as well.
Cannot propose an upgrade without this being solved.
Any progress?

**Comment by entmike** [comment]
Sorry, I am basically the only person maintaining these components in my little spare time anymore.

The LaunchPad uses commons mode components that are not possible in m mode.  This was SAP's decision to obsolete commons mode, nothing I can do about this unless someone can find an m mode equivalent of the TileContainer that has the same functionality.

**Comment by Thirobin** [comment]
Hi Mike, 

Thx for your reply.  However, it's very critical for us too.

After some research, i find this link  which seems to be the TileContainer in M mode : https://sapui5.hana.ondemand.com/sdk/#/sample/sap.m.sample.TileContainer/preview

I'm right or the issue is more complex ?

If you have some technical detail about all of this, could you share them with me ?
We needed to fixe this, even if we have to create a new component.

Best regard,
Thibault

**Comment by entmike** [comment]
I'm sorry, I was actually incorrect and thinking of a different component missing from the m library than the Tile Container.  I wrote this using m mode a few years back (it's missing from commons, so I was backwards on this).

I'll have a look at this one and hopefully it's a simple fix.

**Comment by entmike** [comment]
Ok now I'm really confused, because this is working just fine for me:

![image](https://user-images.githubusercontent.com/174348/31554724-12239a38-b004-11e7-80f2-60122edd5e1d.png)

Can you please confirm you are using the .zip from this repository?  (lumiradesignercommunityext and NOT the one from sdkinstall)

**Comment by Thirobin** [comment]
Hi Mike, 

Thx for your reply. 
But the current problem with Lumira Designer and the Launchpad component is that the scripting method for adding a tile (in execute application) doesn't work : No ones tiles are added to the component 

Scripting method using : LAUNCHPAD_1.addTile("Tile");

However, is working fine when i add tile with the "Additionnal property" of the component. Is only the scripting method that doesn't work.

And i confirm i'm currently using the .ZIp from the lumiradesignercommunityext repository.

Regards,

Thibault

**Comment by entmike** [comment]
Ok, now it is clear to me what the problem is.  I've fixed it.  Please re-download and test again and let me know when you have a chance.

**Comment by dcm4you** [comment]
Hello Mike,

Thanks for your effort! (and Thibault for elaborate the issue further on). On a first peak the 2.0 launchpad 2.0 component is working again (with the script method). Thanks!

With regards

Dennis

**Comment by Thirobin** [comment]
Hello Mike,

Thanks for all, it's working perfectly !

Best regards,

Thibault

**Comment by jboss1995** [comment]
It is still not working for me, am I downloading from the wrong place?

https://github.com/org-scn-design-studio-community/lumiradesignercommunityext/blob/master/releases/org.scn.community.sdk.extensions.zip?raw=true

**Comment by jboss1995** [comment]
It is from the first section:
TL;DR
If you are on Lumira Designer 2.0 SP02, please download updated open source extensions **here**.
If you have a new problem starting in Lumira Designer 2.0 SP02 with the components, report them on GitHub Issues here.
If you are on Design Studio 1.6, you can continue to use your old download link (referenced below.)

**Comment by entmike** [comment]
Yes you download the zip and the zip is what you install (do NOT unzip anything)

**Comment by jboss1995** [comment]
Sure, I get it installed fine, however I'm not able to add tiles programmatically. I have created a simple application with just with just Launch Pad 2.0 extension. I added two tiles manually then added one with the code:

LAUNCHPAD_1.addTile("Sales", "YTD", "sap-icon://money-bills", "1234", "USD", "good");

But only the two I added manually show up. I do get this in the console.  

combined_static_includes_1.js?version=20170802133308:51 [Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
send @ combined_static_includes_1.js?version=20170802133308:51

**Comment by jboss1995** [comment]
entmike, sorry I know you are busy. We are trying to upgrade from BO 4.1 to BO 4.2 and we need to decide if we should use Lumira Design or Design Studio. Would you recommend just going with Design Studio because of this issue?

**Comment by entmike** [comment]
jboss - I thikn I know what's wrong.  Can you remove the final "good" parameter at the end of your statement for now?  I'll get that corrected, removing that parameter for now should get you by.

To answer your question, I'd go with Lumira Designer 2.0.  This issue isn't difficult or big enough to sway such a major decision, IMO.

**Comment by entmike** [comment]
Ok I figured it out.  The code hint window I am give is wrong.  It should be as follows, I will upload a fix today, but in case I get busy, just refer to this, it should work, I simply left out a parameter in front of "good" in the code example, it should be "Success" then "good".  "Success" is a valueState property of which must be one of 4 specific values, otherwise it will fail, which is what is happening because my example was saying "good" is correct, which it is not.  ("good" would be a valid cssClass though.)

![image](https://user-images.githubusercontent.com/174348/31832657-6ac376ca-b58d-11e7-8911-1d7373725a8f.png)

**Comment by jboss1995** [comment]
That did it. Thanks and sorry I should have figured that out myself. I now see what happen, when we promoted it from one server to the other it reset the CSS file to the default one. I did an inspect from the web browser and did not see the tiles at all. Once i changed it to the right CSS file it started working.

Thanks

**Comment by entmike** [comment]
I've pushed this change up with the latest build.  Closing.

**Comment by jboss1995** [comment]
Thanks Mike.
